### PR TITLE
Fix: Mapkit search accuracy

### PR DESCRIPTION
**Pull request by TimBroddin**
Fixes https://github.com/Automattic/dotcom-forge/issues/5998

Adding some before/after screenshots to the original issue.

## Proposed changes:
- Calls the search endpoint after autocomplete, this is more accurate

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Testing instructions:

1. Apply this PR (easiest is to a Simple Site using the instructions below)
2. Add a map block
3. Enjoy the added accuracy

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/map/location-search/mapkit.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ const MapkitLocationSearch = ( { label, onAddPoint } ) => {
 					const withPlaceName = filtered.map( result => ( {
 						...result,
 						placeName: result.displayLines?.join( ', ' ),
+						original: result, // save the original result for later - otherwise mapkit.js gives a type error
```
Comment (mashikag) [inline projects/plugins/jetpack/extensions/blocks/map/location-search/mapkit.js:33]: Interesting solution, but it works! 😁

Comment (TimBroddin) [reply projects/plugins/jetpack/extensions/blocks/map/location-search/mapkit.js:33]: Believe me ... I tried more elegant ways 😁

**Comment by TimBroddin** [comment]
@mashikag Can you approve please? 🙏

**Review by mashikag — APPROVED** [review]
The solution tests well and the code looks good too. 👌

**Pull request merged**